homeassistant/components/camera/foscam.py

```diff
@@ -75,20 +75,32 @@ def motion_detection_enabled(self):
 
     def enable_motion_detection(self):
         """Enable motion detection in camera."""
-        ret, err = self._foscam_session.enable_motion_detection()
-        if ret == FOSCAM_COMM_ERROR:
-            _LOGGER.debug("Unable to communicate with Foscam Camera: %s", err)
-            self._motion_status = True
-        else:
+        try:
+            ret, err = self._foscam_session.enable_motion_detection()
```

Comment:
What's the use of returning an `err` code if it can also throw exceptions 🤔

Reply:
A question to the original author perhaps. This patch is a sticking plaster to fix a long standing bug.